### DISCUSSION TITLE
facing issues while downloading CAMS data

### DISCUSSION THREAD

**Discussion by sghbpp** [pvlib/pvlib-python#1903] [Q&A] [answered]
Unable to download CAMS data.

error - TypeError: cannot unpack non-iterable NoneType object
Code - 

        import pandas as pd 
        import pvlib
        
        latitude = 56.2639 
        longitude = 9.5018  
        start_date_str = '2022-01-07'  
        end_date_str = '2022-01-09' 
        email = 'please use yours'  #  registered email address with SoDa

        # Convert string dates to datetime objects
        start_date = pd.to_datetime(start_date_str)
        end_date = pd.to_datetime(end_date_str)

        # Retrieve the data using the pvlib.iotools.get_cams function
        data, metadata = pvlib.iotools.get_cams(latitude= latitude, longitude= longitude, start = start_date, 
                                                end= end_date, email= email, identifier='cams_radiation',
                                                altitude=None, time_step='1min', time_ref='UT', verbose=False, integrated=False,
                                                label=None, map_variables=True, server='www.soda-is.com', timeout=30)
        # Display the retrieved data and metadata
        print("Data:")
        print(data.head())  # Display the first few rows of the retrieved data
        print("\nMetadata:")
        print(metadata)  # Display the metadata of the requested time-series

error 
Traceback (most recent call last):

  File ~\anaconda3\lib\site-packages\spyder_kernels\py3compat.py:356 in compat_exec
    exec(code, globals, locals)

  File c:\users\............\python codes\iotools\cams\cams.py:23
    data, metadata = pvlib.iotools.get_cams(latitude= latitude, longitude= longitude, start = start_date,

TypeError: cannot unpack non-iterable NoneType object

**Comment by AdamRJensen** [comment, marked as answer]
@sghbpp I am able to reproduce the error when I use an email that is not registered on the SoDa portal. However, when I use a registered email, the code snippet above works just fine. 

Fortunately, the error message handling is currently being updated, which should provide more helpful error messages in the future (see #1799 and #1905)

In the future, please post potential code bugs as [issues](https://github.com/pvlib/pvlib-python/issues) instead of in the discussion section.

**Reply by sghbpp** [reply]
Thanks @AdamRJensen  for your response. 

Actually although i had registered and created an account of soda portal, but i had not accepted their terms and conditions, which i did by logging into their portal. And, now i am able to download the data. Many thanks. 

However, it would be better if we could also include other parameters like temperature, humidity, etc.

**Reply by AdamRJensen** [reply]
@sghbpp the CAMS service does not have those parameters. You may check out pvgis for those or get them straight from ERA5 (although this is a bit tedious at the moment)
